pkg/controller/daemon/controller.go

```diff
@@ -0,0 +1,496 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package daemon
+
+import (
+	"reflect"
+	"sort"
+	"time"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/cache"
+	"k8s.io/kubernetes/pkg/client/record"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/controller"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	"k8s.io/kubernetes/pkg/expapi"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util"
+	"k8s.io/kubernetes/pkg/util/workqueue"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+const (
+	// Daemon sets will periodically check that their daemon pods are running as expected.
+	FullDaemonSetResyncPeriod = 30 * time.Second // TODO: Figure out if this time seems reasonable.
+	// Nodes don't need relisting.
+	FullNodeResyncPeriod = 0
+	// Daemon pods don't need relisting.
+	FullDaemonPodResyncPeriod = 0
+	// If sending a status upate to API server fails, we retry a finite number of times.
+	StatusUpdateRetries = 1
+)
+
+// DaemonSetController is responsible for synchronizing DaemonSets objects stored
+// in the system with actual running pods.
+type DaemonSetController struct {
```

Comment:
This is going to be totally nit-picky, but can you call this DaemonSetsController? That makes it clearer that this controller is managing multiple DaemonSets, not just one.